### PR TITLE
Support manifest filepaths

### PR DESCRIPTION
**Pull request by nolange**
This will add support for mapping paths with the [MANIFESTINPUT Argument](https://learn.microsoft.com/en-us/cpp/build/reference/manifestinput-specify-manifest-input).

### CODE REVIEW

**Comment by mstorsjo** [comment]
This looks reasonable to me - any comments from @huangqinjin?

File: wrappers/wine-msvc.sh [hunk 1]
```diff
@@ -15,49 +15,46 @@
...
-while [ $# -gt 0 ]; do
-	a=$1
-	case $a in
+for a; do
+	path=
+	case "$a" in
 	[-/][A-Za-z]/*)
-		opt=${a:0:2}
-		path=${a:2}
+		path=${a#??}
```
Comment (huangqinjin) [inline wrappers/wine-msvc.sh:26]: any benefit using this form instead of the original one that substring by pos:len ?

**Review by huangqinjin — COMMENTED** [review]
It is acceptable to me. But can we make it better? Firstly, the linker options are case-insensitive. Secondly all linker options are in form of `/OPTION:arg`, we can take this opportunity to handle all linker options.

**Comment by huangqinjin** [comment]
We already have the idea in https://github.com/mstorsjo/msvc-wine/pull/96#issuecomment-1858684632

File: wrappers/wine-msvc.sh (hunk 1 above)
Comment (nolange) [reply wrappers/wine-msvc.sh:26]: Broader compatibility with shells (POSIX standard), otherwise no

**Comment by nolange** [comment]
> It is acceptable to me. But can we make it better? Firstly, the linker options are case-insensitive. Secondly all linker options are in form of `/OPTION:arg`, we can take this opportunity to handle all linker options.

Do we want to handle **all linker options**, by guessing if the argument is a filename? could cause more trouble than what it is worth.

**Comment by huangqinjin** [comment]
> Do we want to handle **all linker options**, by guessing if the argument is a filename? could cause more trouble than what it is worth.

I think we are already doing so. @mstorsjo can you comment on this?

**Comment by mstorsjo** [comment]
> > Do we want to handle **all linker options**, by guessing if the argument is a filename? could cause more trouble than what it is worth.
> 
> I think we are already doing so. @mstorsjo can you comment on this?

Yeah, we already do kind of check whether the argument value seems like a pathname, so I think it might be ok to extend it to a pattern rule like this.

@nolange The updated patch looks good, but you can split it back into two commits - one for factorizing the existing code, and one for adding the new case?

**Comment by mstorsjo** [comment]
Thanks, this looks good to me. I'll merge it later if it passes all the CI checks, unless there's any further opinions on it.

**Pull request merged**